packages/zoe/src/cleanProposal.js

```diff
@@ -52,18 +60,16 @@ const cleanKeys = (allowedKeys, record) => {
 export const getKeywords = keywordRecord =>
   harden(Object.getOwnPropertyNames(keywordRecord));
 
-export const coerceAmountKeywordRecord = (
-  getAmountMath,
-  allegedAmountKeywordRecord,
-) => {
+export const coerceAmountKeywordRecord = allegedAmountKeywordRecord => {
   const keywords = getKeywords(allegedAmountKeywordRecord);
   keywords.forEach(assertKeywordName);
 
   const amounts = Object.values(allegedAmountKeywordRecord);
   // Check that each value can be coerced using the amountMath
   // indicated by brand. `AmountMath.coerce` throws if coercion fails.
+  // TODO: coercing with the amount's same brand is a bit weird. Fix this.
   const coercedAmounts = amounts.map(amount =>
-    getAmountMath(amount.brand).coerce(amount),
+    amountMath.coerce(amount, amount.brand),
```

Comment:
I think I fixed this by having a function `getMathKindByBrand` in Zoe that looks up the mathKind in the issuerTable. It will be deleted when the brand itself has the mathKind property, but should work for now